### PR TITLE
crosschain swaps: watch pending swaps & bridging

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes TEAM2-480
Figma link (if any):

## What changed (plus any additional context for devs)

Integrate backend update to watch crosschain swaps and bridges happening inside the app

## Screen recordings / screenshots

PoW https://www.loom.com/share/c5c50ddb372d44b2b020dc1667e63a96

## What to test

- do swap and bridge like the video and check the transactions is correctly confirmed on ios and android following the video
- killing the app and reopening should keep the watcher updating correctly
- also added a test to do USDC bridging, so i updated the socket watch tx method to return a correct value if IS_TEST

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/redux/data.ts [hunk 1]
```diff
@@ -1580,7 +1542,18 @@ export const checkPendingTransactionsOnInitialize = (
...
-  await dispatch(dataWatchPendingTransactions(provider, currentNonce));
+  const notPendingTxs = await dispatch(
+    dataWatchPendingTransactions(provider, currentNonce)
+  );
+  if (!notPendingTxs) {
+    dispatch(
+      watchPendingTransactions(
+        currentAccountAddress,
+        TXN_WATCHER_MAX_TRIES,
+        null
```
Comment (estebanmino) [inline src/redux/data.ts:1553]: when there's a pending tx on initialize we're currently taking the mainnet provider, which doesn't makes sense for l2 txs, so now we'll be choosing provider depending on the tx network

**Comment by derHowie** [comment]
Bug 1: https://recordit.co/GgEJyukIKK

I shouldn't be able to see the speed-up/cancel options after the tx is confirmed on etherscan, right? In this example I'm seeing that we are detecting transaction confirmation in the logs, but the listener doesn't appear to be removed. The logs indicate we are continually seeing tx confirmation but not appropriately resolving afterwards because the UI appears to stuck in the 'bridging' state.

**Comment by derHowie** [comment]
Bug 2: https://recordit.co/8B3BDS1V9e

I'm seeing some really bizarre behavior when selecting inputs/outputs. In this video the first selection behaves appropriately, but all subsequent selections are placing the wrong tokens in the input/output screen.

**Review by derHowie — COMMENTED** [review]
When bridging and swapping, I see [this line](https://github.com/rainbow-me/rainbow/blob/%40esteban/crosschain-swaps-sdk-swap-status/src/redux/data.ts#L1371) firing repeatedly, however I do not believe that the `'transactionConfirmed'` event is firing. Both transaction cells are appear stuck, although the transactions are going through fine.

https://recordit.co/lvkJ11zt27
https://recordit.co/gbawwkpjIW

**Comment by derHowie** [comment]
Looks pretty good, but I think I found an actual issue:
https://recordit.co/TOzXbJxIWb

Upon bridging completion, I see dupe ETH in the wallet. Guessing that ETH is probably a special case and we need to transform the address before running it through the balance updating functions.

**Comment by estebanmino** [comment]
@derHowie thanks for the review! should be working now, i'm only triggering `getUpdatedL2AssetBalance` for assets that are L2 assets now,

**Review by brunobar79 — APPROVED** [review]
🤝

**Pull request merged**